### PR TITLE
Deactivating, reactivating, and deleting a FusionAuth user in Python

### PR DESCRIPTION
**Pull request by matt1hathcock**
- Updated date added
- Updated site navigation
- Updated title and section headers
- Updated table of contents
- Updated Python version
- Removed Ubuntu reference
- Updated images with class="img-fluid"
- Added Kickstart language
- ID -> Id
- 127.0.0.1 -> localhost, and removed hyperlink
- bash -> zsh
- shell prompts removed
- added hyperlink to "creating-an-api-key"
- "FusionAuth Python client library" where referenced
- other changes

### CODE REVIEW

File: site/_posts/2019-10-08-deactivating-reactivating-deleting-user-fusionauth-python.md [hunk 1]
```diff
@@ -128,22 +137,22 @@ client = FusionAuthClient('YOUR-OWN-API-KEY', 'http://localhost:9011')
...
 ```
 
-Save the program to your server, make sure that `venv` is running, and execute the program with `deactivate.py`. If successful, you'll see a response of `None`.
+Save the program to your server, make sure that `venv` is running, and execute the application `deactivate.py`. If successful, you'll see a response of `None`.
 
-Now, let's head on over to a web browser, and browse to **Users** so that we can check out our user. You should notice that the account associated with the UserID we provided is now locked. It will look something like this:
+Now, let's head on over to a web browser, and browse to "Users" so that we can check out our user. You should notice that the account associated with the user Id we provided is now locked. It will look something like this:
 
-{% include _image.liquid src="/assets/img/blogs/terry-example-locked.png" alt="Terry is now locked" class="img-fluid mb-4 mt-2 mx-auto d-block" figure=false %}
+{% include _image.liquid src="/assets/img/blogs/terry-example-locked.png" alt="Terry is now locked" class="img-fluid" figure=false %}
```
Comment (mooreds) [inline site/_posts/2019-10-08-deactivating-reactivating-deleting-user-fusionauth-python.md:150]:
```suggestion
When a user is locked, they can no longer log in to any applications, but they are not removed from the system. This is useful if you want to re-enable them later (perhaps after they've paid their bill).

```

File: site/_posts/2019-10-08-deactivating-reactivating-deleting-user-fusionauth-python.md [hunk 2]
```diff
@@ -152,22 +161,22 @@ from fusionauth.fusionauth_client import FusionAuthClient
...
+  print(client_response.success_response)
 else:
-	print(client_response.error_response)
+  print(client_response.error_response)
 ```
 
 Save and execute the `reactivate.py` application on your server. Upon success, you'll get a client response that should look something like this, displaying the user's profile information:
 
-```bash
+```zsh
```
Comment (mooreds) [inline site/_posts/2019-10-08-deactivating-reactivating-deleting-user-fusionauth-python.md:179]: Isn't this python?

File: site/_posts/2019-10-08-deactivating-reactivating-deleting-user-fusionauth-python.md [hunk 3]
```diff
@@ -205,72 +214,72 @@ Save and execute the `reactivate.py` application on your server. Upon success, y
...
+Refresh the "Users" page in your browser and you'll now notice that the lock icon has disappeared beside our test user and the account has been reactivated.
 
-{% include _image.liquid src="/assets/img/blogs/terry-example-unlocked.png" alt="Terry is now unlocked" class="img-fluid mb-4 mt-2 mx-auto d-block" figure=false %}
+{% include _image.liquid src="/assets/img/blogs/terry-example-unlocked.png" alt="Terry is now unlocked" class="img-fluid" figure=false %}
 
-## How to Delete A User With FusionAuth-Client
+## How to delete a user with FusionAuth Python client library
 
-For our next trick, we will be deleting a test user from the server. If you filled out a lot of profile information on your regular test user, you might want to create an empty test user for this tutorial. Just be sure to grab their UserID from the **Actions-->Manage** section of FusionAuth.
+For our next trick, we will be deleting a test user from the server. If you filled out a lot of profile information on your regular test user, you might want to create an empty test user for this tutorial. Just be sure to grab their "User Id" from the "Manage User" section of FusionAuth.
```
Comment (mooreds) [inline site/_posts/2019-10-08-deactivating-reactivating-deleting-user-fusionauth-python.md:223]: Why would you delete a user vs deactivate them? Did that get covered above (might have missed it).

File: site/_posts/2019-10-08-deactivating-reactivating-deleting-user-fusionauth-python.md [hunk 4]
```diff
@@ -205,72 +214,72 @@ Save and execute the `reactivate.py` application on your server. Upon success, y
...
 
-Pop open your Python editor, and create a new file called `deleteuser.py`. Add the FusionAuth_client Library, Your API key, and your server info.
+Pop open your Python editor, and create a new file called `delete.py`. Import the FusionAuth client library, your API key, and your server info.
 
 ```python
 from fusionauth.fusionauth_client import FusionAuthClient
 
-#  You must supply your API key and URL here
+# You must supply your API key and URL here
 client = FusionAuthClient('YOUR-API', 'http://localhost:9011')
```
Comment (mooreds) [inline site/_posts/2019-10-08-deactivating-reactivating-deleting-user-fusionauth-python.md:233]:
```suggestion
client = FusionAuthClient('YOUR-API-KEY', 'http://localhost:9011')
```

File: site/_posts/2019-10-08-deactivating-reactivating-deleting-user-fusionauth-python.md [hunk 5]
```diff
@@ -205,72 +214,72 @@ Save and execute the `reactivate.py` application on your server. Upon success, y
...
 else:
-	print(client_response.error_response)
+  print(client_response.error_response)
 ```
 
-Save and execute your `deleteuser.py` application. (**Caution:** Executing this action will actually delete the user.)
+Save and execute your `delete.py` application. (**Caution:** Executing this action will actually delete the user.)
 
-If successful, you will get a response of `None` and the user will now be deleted from FusionAuth. You can test this again by refreshing the **Users** page in your browser.  
+If successful, you will get a response of `None` and the user will now be deleted from FusionAuth. You can test this again by refreshing the "Users" page in your browser.
```
Comment (mooreds) [inline site/_posts/2019-10-08-deactivating-reactivating-deleting-user-fusionauth-python.md:249]: Hmmm. Is this a hard delete or a soft delete? (Hard delete actually removes them from the system, soft delete lets you restore later.)  Looking at the source, it looks like a hard delete: https://github.com/FusionAuth/fusionauth-python-client/blob/master/src/main/python/fusionauth/fusionauth_client.py

File: site/_posts/2019-10-08-deactivating-reactivating-deleting-user-fusionauth-python.md [hunk 6]
```diff
@@ -1,120 +1,129 @@
...
-1. How to Delete a User
-1. Troubleshooting and Common Errors
+**Most recently updated July 23, 2021.**
+
+## What we'll cover
+
+1. Introduction to FusionAuth user management in Python
+1. Verifying server applications, libraries, and dependencies
+1. Obtaining a test user account Id
+1. How to deactivate a user
```
Comment (mooreds) [inline site/_posts/2019-10-08-deactivating-reactivating-deleting-user-fusionauth-python.md:24]: Did you add any sections? I think so (kickstart)

File: site/_posts/2019-10-08-deactivating-reactivating-deleting-user-fusionauth-python.md [hunk 7]
```diff
@@ -1,120 +1,129 @@
...
-On the **Manage User** page in the FusionAuth UI, you'll see the user ID located directly under the user's Email address. In this instance, it's `4e0f2af0-33a2-4001-8631-0f8225e65f1c`.
+On the "Manage User" page in the FusionAuth UI, you'll see the "User Id" located directly under the user's email address. In this instance, it's `4e0f2af0-33a2-4001-8631-0f8225e65f1c`.
 
 Copy it into a text editor and set it off to the side. We'll be using it in a moment and throughout the rest of the tutorial.
 
-{% include _image.liquid src="/assets/img/blogs/terry-test-account.png" alt="Terry test user info" class="img-fluid mb-4 mt-2 mx-auto d-block" figure=false %}
+{% include _image.liquid src="/assets/img/blogs/terry-test-account.png" alt="Terry test user info" class="img-fluid" figure=false %}
 
-## How to Deactivate a User with FusionAuth-Client
+## Setting up FusionAuth using a Kickstart file
```
Comment (mooreds) [inline site/_posts/2019-10-08-deactivating-reactivating-deleting-user-fusionauth-python.md:97]: I'd add a section here defining each of these terms. Something like this (but jazz it up if you want).

* deactivate, also known as a soft delete or account locking, lets you disable the ability for a user to login temporarily
* reactivate removes the lock and lets a user login
* delete, also known as a hard delete, removes all data for a user and can't be undone. Beware!

File: site/_posts/2019-10-08-deactivating-reactivating-deleting-user-fusionauth-python.md [hunk 8]
```diff
@@ -286,25 +295,25 @@ ImportError: No module named fusionauth.fusionauth_client
...
-Didn't find your error here? Visit [https://fusionauth.i/docs/v1/tech/apis/users](/docs/v1/tech/apis/users) for more response codes, or visit [https://fusionauth.io/docs/v1/tech/troubleshooting](/docs/v1/tech/troubleshooting) for more troubleshooting suggestions. You can also ask a question on [StackOverflow](https://stackoverflow.com/questions/tagged/fusionauth) or open an issue on [Github](https://github.com/FusionAuth/fusionauth-issues/issues/new/choose). As always, if you have a support contract feel free to [contact FusionAuth support](mailto:support@fusionauth.io).
+### Additional error codes
+Didn't find your error here? Visit [https://fusionauth.i/docs/v1/tech/apis/users](/docs/v1/tech/apis/users) for more response codes, or visit [https://fusionauth.io/docs/v1/tech/troubleshooting](/docs/v1/tech/troubleshooting) for more troubleshooting suggestions. You can also ask a question on the [FusionAuth Forum](https://fusionauth.io/community/forum/) or open an issue on [Github](https://github.com/FusionAuth/fusionauth-issues/issues/new/choose). As always, if you have a support contract feel free to [contact FusionAuth support](mailto:support@fusionauth.io).
```
Comment (mooreds) [inline site/_posts/2019-10-08-deactivating-reactivating-deleting-user-fusionauth-python.md:309]:
```suggestion
Didn't find your error here? Visit [https://fusionauth.i/docs/v1/tech/apis/users](/docs/v1/tech/apis/users) for more response codes, or visit [https://fusionauth.io/docs/v1/tech/troubleshooting](/docs/v1/tech/troubleshooting) for more troubleshooting suggestions. You can also ask a question on the [FusionAuth Forum](https://fusionauth.io/community/forum/). As always, if you have a support contract feel free to [contact FusionAuth support](https://account.fusionauth.io/account/support).
```

**Review by mooreds — CHANGES_REQUESTED** [review]
A few changes.

Meta comments:

* Did you run this through google docs for a grammar check?
* David is going to create a new header image, will share when that is done
* what version of fusionauth did you use for the update? Please add a call out like you did for the other tutorial
* is python-first-app the correct kickstart file? Wasn't sure if there was another one.

File: site/_layouts/blog-post.liquid [hunk 9]
```diff
@@ -49,7 +49,7 @@
         <article class="col-xs-12 col-lg-9">
           <img class="post-image" src="/assets/img/{{ page.image }}" alt="{{ page.description }}"/>
           <div class="info">
-            <div class="author-date">{{ page.author }}&nbsp;&nbsp;|&nbsp;&nbsp;{{ page.date | date_to_string: "ordinal", "US" }} {%if page.updated_date%}| Most recently updated: {{ page.updated_date | date_to_string: "ordinal", "US" }}{%endif%}</div>
+            <div class="author-date">{{ page.author }}&nbsp;&nbsp;|&nbsp;&nbsp;{{ page.date | date_to_string: "ordinal", "US" }}{%if page.updated_date%}&nbsp;&nbsp;|&nbsp;&nbsp;Most recently updated: {{ page.updated_date | date_to_string: "ordinal", "US" }}{%endif%}</div>
```
Comment (mooreds) [inline site/_layouts/blog-post.liquid:52]: This was me, fyi.

File: site/_posts/2019-10-08-deactivating-reactivating-deleting-user-fusionauth-python.md (hunk 7 above)
Comment (mooreds) [reply site/_posts/2019-10-08-deactivating-reactivating-deleting-user-fusionauth-python.md:102]: looks good now.

File: site/_posts/2019-10-08-deactivating-reactivating-deleting-user-fusionauth-python.md (hunk 2 above)
Comment (mooreds) [reply site/_posts/2019-10-08-deactivating-reactivating-deleting-user-fusionauth-python.md:194]: please change it to python.

**Comment by mooreds** [comment]
One change and then merge away! https://github.com/FusionAuth/fusionauth-site/pull/837/files#r708694133

File: site/_posts/2019-10-08-deactivating-reactivating-deleting-user-fusionauth-python.md (hunk 2 above)
Comment (matt1hathcock) [reply site/_posts/2019-10-08-deactivating-reactivating-deleting-user-fusionauth-python.md]: fixed in latest commit

**Pull request merged**